synapse/http/site.py

```diff
@@ -500,6 +501,23 @@ def _should_log_request(self) -> bool:
 
         return True
 
+    def get_client_ip_if_available(self) -> str:
+        """Logging helper. Return something useful when a client IP is not retrievable
+        from a unix socket.
+
+        In practice, this returns the socket file path on a SynapseRequest if using a
+        unix socket and the normal IP address for TCP sockets.
```

Comment:
This seems like a *major* caveat to using unix sockets if you can't get the original IP address of the requester?

Reply:
It feels very wrong, but consider that it has it's usage even if the concept seems wrong and foreign. Unix sockets have no concept of the things we are used to seeing, like IP addresses and ports and even TLS and SSL. It knows where to send the result of a request, but for us to be able to see and log it, we like to see something familiar and almost intuitively known. So I put the effort forward to pull that data out in cases where I could reliably get it and at the minimum at least *something* when I can't. Other than logs, it's not really necessary stuff 🤷‍♂️

Reply:
the UNIX socket is going to be used with a reverse proxy anyway, in which case you need to rely on `X-Forwarded-For` as you would today if you used TCP. This is probably overcomplicating things and should maybe just return something equivalent to `"n/a"`